### PR TITLE
feat: Detect stringified json object and array as OBJECT and ARRAY type

### PR DESCRIPTION
**Pull request by heshan0131**
- test stringified json object as OBJECT type
- test stringified array as ARRAY type

### CODE REVIEW

**Review by btford — APPROVED** [review]
LGTM

File: src/utils.js [hunk 1]
```diff
@@ -48,9 +48,67 @@ function whichFormatGenerator(formatRegex, regexMap) {
...
+  return parsed;
+}
+
+function isString(value) {
+  return typeof value === 'string';
+}
+
+function isObject(value) {
+  return (
+    value === Object(value) &&
```
Comment (nrabinowitz) [inline src/utils.js:81]: TIL

**Review by nrabinowitz — APPROVED** [review]
LGTM, but it seems like this is a breaking change - if a consumer gets type `ARRAY` or `OBJECT` but doesn't know that they need to parse, won't that be a problem?

**Comment by heshan0131** [comment]
> LGTM, but it seems like this is a breaking change - if a consumer gets type `ARRAY` or `OBJECT` but doesn't know that they need to parse, won't that be a problem?

It's probably a minor breaking change. we can do a minor version bump and add to release note

**Review by chrisirhc — APPROVED** [review]

**Pull request merged**